apps/wallet-api-tools/src/Editor.tsx

```diff
@@ -1,92 +1,159 @@
 "use client";
-
+import { useAtom } from "jotai";
+import { atomWithStorage } from "jotai/utils";
 import { Transport, WindowMessageTransport } from "@ledgerhq/wallet-api-core";
 import {
   getSimulatorTransport,
   profiles,
 } from "@ledgerhq/wallet-api-simulator";
-import { langs } from '@uiw/codemirror-extensions-langs';
-import CodeMirror, { ReactCodeMirrorRef } from "@uiw/react-codemirror";
 import { useSearchParams } from "next/navigation";
-import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { useCallback, useEffect, useRef, useState } from "react";
 import { v4 as uuidv4 } from "uuid";
 import { z } from "zod";
 import { Input } from "./Input";
+import { MessageGrouped, Message } from "./types";
+import GroupedMessage from "./components/GroupedMessage";
+import InfoMessage from "./components/InfoMessage";
+import { ArrowDown } from "lucide-react";
 
 const schemaRequest = z.object({
   method: z.string(),
   params: z.object({}).passthrough(),
 });
 
-type Message = {
-  type: "in" | "out" | "info" | "error";
-  value: string;
-  date: Date;
-};
-
-function getMessageStatus(message: Message) {
-  switch (message.type) {
-    case "in":
-      return "<- [received]";
-    case "out":
-      return "-> [sent]";
-    case "error":
-      return "[error]";
-    case "info":
-      return "[info]";
-    default: {
-      const exhaustiveCheck: never = message.type; // https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking
-      return exhaustiveCheck;
-    }
-  }
-}
+const historyAtom = atomWithStorage<(MessageGrouped | Message)[]>(
+  "history",
+  [],
+);
 
 export function Editor() {
-  const [history, setHistory] = useState<Message[]>([]);
+  const [history, setHistoryAtom] = useAtom(historyAtom);
```

Comment:
We can probably keep the `setHistory` name instead of `setHistoryAtom`
```suggestion
  const [history, setHistory] = useAtom(historyAtom);
```